### PR TITLE
grpclog: implement Infof on V

### PR DESCRIPTION
**Pull request by menghanl**
https://github.com/grpc/grpc-go/issues/2709#issuecomment-475546809

### CODE REVIEW

File: grpclog/grpclog.go [hunk 1]
```diff
@@ -30,9 +30,24 @@ import "os"
 
 var logger = newLoggerV2()
 
+// Verbose is a typed boolean that implements Infof.
+type Verbose bool
+
+// Infof is equivalent to the global Infof, guarded by the verbose value.
+func (v Verbose) Infof(format string, args ...interface{}) {
```
Comment (Capstan) [inline grpclog/grpclog.go]: Do you also want to provide bare `Info`?

Comment (menghanl) [reply grpclog/grpclog.go]: Added both `Info` and `Infoln`.

I was thinking `Infof` should work for all the cases.

**Comment by dfawley** [comment]
I think we don't need this.  It could be harmful as well.  E.g. in the event of usage such as this:

```go
grpclog.V(2).Infof("something: %v", expensiveFunction())
```

`expensiveFunction` will be called unconditionally, even when `V<2`.

**Comment by Capstan** [comment]
@dfawley While true, one can still use the older if style to selectively calculate expensive things, though now it becomes incumbent on authors to evaluate that. This style also matches the `V()` behavior used in the standard logging package used in the Google mono repo since that package's inception in April 2012, so its tires have been kicked for some time.

**Comment by dfawley** [comment]
:man_shrugging: I don't have super strong feelings on this either way.  Given that the grpc-go devs don't intend to use this functionality any time soon, let's leave it out until it seems important.  It's worth mentioning that this package is externally visible, even though it was intended for grpc-internal use only.  We should drop a note in the package comments specifying this intent, but we can't really move it to `internal` to avoid a breaking change.

**Pull request closed without merging**